torchinductor/utils.py

```diff
@@ -165,3 +165,39 @@ def wrapper(self):
         return getattr(self, key)
 
     return wrapper
+
+
+def has_incompatible_cudagraph_ops(gm):
+    forbidden_list = set(
+        [
+            "fbgemm.dense_to_jagged.default",
+            "fbgemm.jagged_to_padded_dense.default",
+        ]
+    )
+    for node in gm.graph.nodes:
+        if str(node.target) in forbidden_list:
+            return True
+    return False
+
+
+def assert_size_match(item, size):
+    assert isinstance(size, tuple)
+    if size == ():
+        return
+
+    if isinstance(item, (list, tuple)):
```

Comment:
If we are calling this function then inductor thinks `item` is a tensor.  This is clearly wrong, so I think this isn't fixing the underlying issue (which is TorchInductor got the type wrong).

Reply:
Yes - but perhaps instead of flattening, we need to keep the same top-level structure but then turn the nested contents into data that is a list or tuple of tensors, instead of flattening here. Let me try that.

Reply:
The problem here, I think, is that the contents of `example_output` is a set, but the size() per element is not uniform in our case. Let me think about this further. Jagged tensor breaks a lot of our `FixedLayout` assumptions. We may need something like JaggedLayout where we support indexing into this more complicated list of sizes in the shape.

Reply:
Or, we unflatten up there, and then maybe make StorageBox take a list...

Let me mull this over. I'd like to represent the data as correctly as possible here.